### PR TITLE
Ba meyer master

### PR DESCRIPTION
**Pull request by lqwert**
Current state of my bachelor thesis with description on how to start the auth server. And code comments on how to turn on/off those capabilities

### CODE REVIEW

File: docs/dev/index.md [hunk 1]
```diff
@@ -22,16 +21,17 @@ This document provides an index to all development guidelines and background inf
 5. Insert the following commands  `docker-compose up -d`
 
 #### IntelliJ
-3. [Follow PatternPediaAPI from Step 5 pls](#step5) 
+3. [Follow PatternPediaAPI from Step 5 pls](#step5)
 
+#### Turn On/Off
```
Comment (manuwei) [inline docs/dev/index.md:26]: Turn On/Off for what? --> I think the new explanations belong to PatternPedia Auth? 
Please also add a few sentences that explain what this component is about

File: src/main/java/com/patternpedia/api/entities/shared/AuthorConstant.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.patternpedia.api.entities.shared;
+
+public interface AuthorConstant {
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/entities/shared/AuthorConstant.java:3]: Why is this an interface and not just a global constant?

File: src/main/java/com/patternpedia/api/entities/user/UserEntity.java [hunk 3]
```diff
@@ -20,29 +24,35 @@
 @Entity
 @Data
 @NoArgsConstructor
-@TypeDef(name = "pgsql_enum", typeClass = PostgreSQLEnumType.class)
+//@NamedQuery(name = "UserEntity.findAll", query="select u from UserEntity u order by u.role.name")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/entities/user/UserEntity.java:27]: is this line stil needed? (By the way, I'd prefer to collect all queries in the service that interacts with the repository - in the best case, you can use the repository functionalities thatare provided by Spring)

File: src/main/java/com/patternpedia/api/entities/user/UserEntity.java [hunk 4]
```diff
@@ -69,17 +83,23 @@
 //    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/entities/user/UserEntity.java:83]: why are the comments deleted?

File: src/main/java/com/patternpedia/api/entities/user/role/PrivilegeConstant.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.patternpedia.api.entities.user.role;
+
+public interface PrivilegeConstant {
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/entities/user/role/PrivilegeConstant.java:3]: using an interface is again not ideal

File: src/main/java/com/patternpedia/api/rest/controller/AuthorController.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    ) {
+        this.authorService = authorService;
+        this.userService = userService;
+    }
+
+    /**
+     * GET Methods
+     */
+    @GetMapping(value = "")
+//    @PreAuthorize(value = "hasAuthority('ADMIN')")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/AuthorController.java:41]: should these lines also be commented out if no authentication is used or is this line of code already obsolete?

File: src/main/java/com/patternpedia/api/rest/model/user/RoleModel.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    private UUID id;
+    private String name;
+//    private Collection<PrivilegeModel> privileges;
+    private Collection<String> privileges;
+
+    public RoleModel (Role role) {
+        this.id = role.getId();
+        this.name = role.getName();
+//        this.privileges = role.getPrivileges().stream().map(privilege -> new PrivilegeModel(privilege)).collect(Collectors.toList());
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/model/user/RoleModel.java:25]: to be deleted lines?

File: src/main/java/com/patternpedia/api/service/AuthorService.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+public interface AuthorService {
+
+    IssueAuthor saveIssueAuthor(UUID userId, UUID issueId, AuthorModelRequest authorModelRequest);
+    void deleteIssueAuthor(UUID userId, UUID issueId);
+
+    CandidateAuthor saveCandidateAuthor(UUID userId, UUID candidateId, AuthorModelRequest authorModelRequest);
+    void deleteCandidateAuthor(UUID userId, UUID candidateId);
+
+//    Candidate updateCandidateAuthor(UUID candidateId, AuthorModelRequest authorModelRequest);
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/service/AuthorService.java:23]: why //?

File: src/main/java/com/patternpedia/api/entities/user/UserEntity.java (hunk 4 above)
Comment (lqwert) [reply src/main/java/com/patternpedia/api/entities/user/UserEntity.java:82]: Which comments are deleted?

File: docs/dev/index.md (hunk 1 above)
Comment (lqwert) [reply docs/dev/index.md]: I added a short description and changed the on/off so it hopefully makes more sense

File: src/main/java/com/patternpedia/api/entities/user/UserEntity.java (hunk 4 above)
Comment (manuwei) [reply src/main/java/com/patternpedia/api/entities/user/UserEntity.java:82]: why is there a '//' on this line? can we delete the line or will we need it in the future?

**Comment by manuwei** [comment]
@lqwert can you check the warning issue? otherwise it looks fine :)

**Pull request closed without merging**